plugins/builds/index.js

```diff
@@ -203,7 +193,7 @@ async function triggerNextJobs(config, app) {
 
             try {
                 if (isOrTrigger(externalEvent.workflowGraph, remoteTriggerName, nextJobName)) {
-                    await remoteTrigger.run(externalEvent, nextJobName, nextJobId, parentBuilds);
+                    await OrTrigger.run(externalEvent, externalEvent.pipelineId, nextJobName, nextJobId, parentBuilds);
```

Comment:
If we use the same class, there is a risk that a if statements (e.g. `if (isRemoteTrigger)`) will be added in the class when a different behavior is needed in RemoteTrigger in the future.
Then codes are getting mixed up, making it difficult to make independent modifications.

I think it is better to create a base class like join case and implement it in the form of inheritance to allow flexibility in the future.

Reply:
Created OrBase class and created OrTrigger and remoteTrigger classes that inherit from it.